pydis_site/apps/api/models/bot/metricity.py

```diff
@@ -89,3 +91,41 @@ def total_message_blocks(self, user_id: str) -> int:
             raise NotFound()
 
         return values[0]
+
+    def top_channel_activity(self, user_id: str) -> List[Tuple[str, int]]:
+        """
+        Query the top three channels in which the user is most active.
+
+        Help channels are grouped under "the help channels",
+        and off-topic channels are grouped under "off-topic".
+        """
+        self.cursor.execute(
+            """
+            SELECT
+                CASE
+                    WHEN channels.name ILIKE 'help-%%' THEN 'the help channels'
+                    WHEN channels.name ILIKE 'ot%%' THEN 'off-topic'
+                    ELSE channels.name
+                END,
+                COUNT(1)
+            FROM
+                messages
+                LEFT JOIN channels ON channels.id = messages.channel_id
+            WHERE
+                author_id = '%s'
```

Comment:
Do we want to include deleted messages (probably a minor thing)?

If not:
```suggestion
                author_id = '%s' AND NOT messages.is_deleted
```

Otherwise, please ignore.